lib/src/Base/Stat/HMatrixImplementation.cxx

```diff
@@ -313,15 +321,17 @@ HMatrixImplementation::assemble(const HMatrixRealAssemblyFunction& f, char symme
   else if (compressionMethod == "AcaRandom")
     ctx_assemble.compression = hmat_create_compression_aca_random(assemblyEpsilon);
   else
-    LOGWARN( OSS() << "Unknown compression method: " << compressionMethod << ". Valid values are: Svd, AcaFull, AcaPartial or AcaPlus");
+    throw InvalidArgumentException(HERE) << "Unknown compression method: " << compressionMethod << ". Valid values are: Svd, AcaFull, AcaPartial, AcaPlus or AcaRandom";
 
-  static_cast<hmat_interface_t*>(hmatInterface_.get())->assemble_generic(static_cast<hmat_matrix_t*>(hmat_), &ctx_assemble);
+  int rc = static_cast<hmat_interface_t *>(hmatInterface_.get())->assemble_generic(static_cast<hmat_matrix_t *>(hmat_), &ctx_assemble);
+  if (rc != 0)
+    throw InvalidArgumentException(HERE) << "Something went wrong in assemble";
```

Comment:
we usually throw InternalException for errors in third-party libs
you could mention hmat in the exception message

Reply:
ok you are right. I will fix it soon